### PR TITLE
Filters plugin URLs to use sandboxed hostnames.

### PR DESCRIPTION
**Pull request by emrikol**
## Description

Adds a `wpvip_filter_sandbox_plugins_url` filter to the `plugins_url` hook to properly return sandboxed hostnames.

Currently `plugins_url( 'test' );` will return a URL like `https://example.com/wp-content/plugins/test` which does not allow for enqueuing assets that are sandboxed.

After the filter, it should properly return URLs like `https://example-com.sandbox-ingress-hostname.example.com/wp-content/plugins/test`

### Plugin Updated: vip-helpers

* Fixes a bug where incorrect URLs are returned by `plugins_url()` when sites are sandboxed by VIP support.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

### CODE REVIEW

File: vip-helpers/sandbox.php [hunk 1]
```diff
@@ -88,3 +88,21 @@ function wpcom_do_sandbox_bar() {
...
+
+/**
+ * Filters plugin URLs to use sandboxed hostnames.
+ *
+ * @param string $url The complete URL to the plugins directory including scheme and path.
+ *
+ * @return string Filtered URL using sandboxed hostname.
+ */
+function wpvip_filter_sandbox_plugins_url( $url ) {
+	if ( defined( 'WPCOM_SANDBOXED' ) && WPCOM_SANDBOXED ) {
```
Comment (rebeccahum) [inline vip-helpers/sandbox.php:100]: We don't need this check since the file returns if the constant isn't defined. Other than that, tested and LGTM.

**Review by rebeccahum — CHANGES_REQUESTED** [review]

**Review by rebeccahum — CHANGES_REQUESTED** [review]
Actually, I tested this further and in multi sites where the site URL contains another site URL, we can have search-replace overlapping.

e.g. subsite 1 is example.com and subsite 2 is example.com/test

Say we want to look at the plugin_url for subsite 2, it gets replaced with subsite 1 first.

File: vip-helpers/sandbox.php [hunk 2]
```diff
@@ -88,3 +88,21 @@ function wpcom_do_sandbox_bar() {
...
+		global $sandbox_vhosts;
+
+		$url = str_replace( array_values( $sandbox_vhosts ), array_keys( $sandbox_vhosts ), $url );
```
Comment (rinatkhaziev) [inline vip-helpers/sandbox.php:103]:
```suggestion
		$url = str_replace( $_SERVER['HTTP_HOST'], array_flip( $sandbox_vhosts )[ $_SERVER['HTTP_HOST'] ], $url );
```

File: vip-helpers/sandbox.php (hunk 1 above)
Comment (rinatkhaziev) [reply vip-helpers/sandbox.php:100]: I think this should need to be swapped with the check for the old platform?

**Review by rinatkhaziev — DISMISSED** [review]
Thanks! Left a bit of inline feedback

**Comment by sjinks** [comment]
@rebeccahum re: overlapping - could you please check the update implementation? It makes sure that we only replace the matching part in the URL domain.

**Review by rebeccahum — APPROVED** [review]

**Comment by rebeccahum** [comment]
Tested & works after the most recent change.

**Comment by rebeccahum** [comment]
r2248-stacks

**Pull request merged**